src/backend/core/models.py

```diff
@@ -284,8 +284,27 @@ def __str__(self):
         return f"{id_str:s}{main_str:s}"
 
     def save(self, *args, **kwargs):
-        """Ensure users always have one and only one main identity."""
+        """
+        Save identity, ensuring user always have one and only one main identity.
+        Also convert valid invitations if new identity.
+        """
+
+        # If new identity, convert all valid invitations to team accesses.
+        # Expired invitations are ignored.
+        if self._state.adding is True:
+            validity_duration = timedelta(seconds=settings.INVITATION_VALIDITY_DURATION)
+            active_invitations = Invitation.objects.filter(
+                email=self.email, created_at__gte=timezone.now() - validity_duration
+            )
```

Comment:
I know `@property` are not accessible when using filter, however this feel duplicating a critical logic. I have better solution in mind, I just wanted to discuss it, anyway feel free to ingore my comment